kroxylicious-operator/src/test/java/io/kroxylicious/kubernetes/operator/OperatorMainTest.java

```diff
@@ -0,0 +1,196 @@
+        expectWatchOn("/apis/kroxylicious.io/v1alpha1/virtualkafkaclusters", VirtualKafkaCluster.class, "virtualkafkacluster", "kroxylicious.io/v1alpha1", 1000);
+        expectWatchOn("/apis/filter.kroxylicious.io/v1alpha1/kafkaprotocolfilters", KafkaProtocolFilter.class, "kafkaprotocolfilter", "kroxylicious.io/v1alpha1", 0);
+        expectWatchOn("/apis/apps/v1/deployments", Deployment.class, "deployment", "v1", 1000);
+        expectWatchOn("/api/v1/secrets", Secret.class, "secret", "v1", 1000);
+        expectWatchOn("/api/v1/services", Service.class, "service", "v1", 1000);
+
+        operatorMain = new OperatorMain(() -> globalRegistry, kubeClient);
+    }
+
+    @Test
+    void shouldRegisterPrometheusMeterRegistry() {
+        // Given
+
+        // When
+        operatorMain.run();
+
+        // Then
+        verify(globalRegistry).add(argThat(PrometheusMeterRegistry.class::isInstance));
+        mockServer.shutdown();
+    }
+
+    @Test
+    void shouldRegisterOperatorMetrics() {
+        // Given
+
+        // When
+        operatorMain.run();
+
+        // Then
+        Awaitility.await()
+                .atMost(10, TimeUnit.SECONDS)
+                .ignoreException(MeterNotFoundException.class)
+                .untilAsserted(() -> assertThat(operatorMain.getRegistry().get("operator.sdk.events.received").meter().getId()).isNotNull());
+
+    }
+
+    @Test
+    void shouldRegisterMetricsForProxyReconciler() {
+        // Given
+
+        // When
+        operatorMain.run();
+
+        // Then
+        assertThat(operatorMain.getRegistry().get("operator.sdk.reconciliations.executions.proxyreconciler").meter().getId()).isNotNull();
+    }
+
+    private void expectApiResources() {
+        final APIResource kafkaProtocolFilterCrd = new APIResourceBuilder()
+                .withCategories("kroxylicious-plugins")
+                .withKind("KafkaProtocolFilter")
+                .withName("kafkaprotocolfilters")
+                .withNamespaced(true)
+                .withShortNames("kpf")
+                .withSingularName("kafkaprotocolfilter")
+                .withVerbs("delete", "deletecollection", "get", "list", "patch", "create", "update", "watch")
+                .build();
+
+        final APIResourceList apiResourceList = new APIResourceListBuilder()
+                .withGroupVersion("filter.kroxylicious.io/v1alpha1")
+                .withApiVersion("v1")
+                .addToResources(kafkaProtocolFilterCrd)
+                .build();
+
+        mockServer.expect()
+                .get()
+                .withPath("/apis/filter.kroxylicious.io/v1alpha1")
+                .andReturn(HttpURLConnection.HTTP_OK, apiResourceList)
+                .always();
+
+        mockServer.expect()
+                .get()
+                .withPath("/apis/kroxylicious.io/v1alpha1/namespaces/randomNs/kafkaproxies")
```

Comment:
Are these versions correct? More generally, maybe we want to factor them into constants to make it easier when we transition `v1alpha1` -> `v1beta1` -> `v1`?

Reply:
They are correct now. I want to take another pass at the unit test now I have something comparable to the integration test. I think `crudMode = true` in `@EnableKubernetesMockClient()` might simplify this but I was tripping over the informers when trying to write them initially.

Reply:
Paths have gone with the new simpler test.